assets/js/components/module/ModuleApp.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * ModuleApp component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { _x, sprintf, __ } from '@wordpress/i18n';
+import { Fragment } from '@wordpress/element';
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { isFeatureEnabled } from '../../features';
+import Header from '../Header';
+import Alert from '../Alert';
+import PageHeader from '../PageHeader';
+import ModuleFooter from './ModuleFooter';
+import LegacyModuleApp from './LegacyModuleApp';
+import WidgetContextRenderer from '../../googlesitekit/widgets/components/WidgetContextRenderer';
+
+import DateRangeSelector from '../DateRangeSelector';
+import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
+
+const { useSelect } = Data;
+
+function ModuleApp( { moduleSlug } ) {
```

Comment:
This is a confusing prop name, because it isn't a module slug: it's an object with the module's `slug` and `name` attributes as properties.

Can we rename this to something more representative? Or we could have `slug` and `name` be props.

Reply:
I agree, this could be clearer. The IB was particular about it:

> - Introduce a new ModuleApp functional hook-based component in assets/js/components/module:
>    - Require moduleSlug as prop.

Looks like that might be to keep parity with the other component that could be rendered in [assets/js/googlesitekit-module.js](https://github.com/google/site-kit-wp/blob/develop/assets/js/googlesitekit-module.js#L43)?

I like the idea of specific props, so let's go with `moduleName` and `moduleSlug`?

Reply:
We're already changing the API anyway so I don't think we need to keep the `moduleName` prop around, it's fine to update this and accept the slug only.

I might just use `slug` as the prop name given the component itself is called `ModuleSetup`, so `slug` implies a module `slug`.

But I think think we should remove the `moduleName`/`name` prop entirely, because the component can select the module name based on the supplied slug:

```js
const module = useSelect( ( select ) => select( CORE_MODULES ).getModule( slug ) );
const { name } = module;
```

I'm a fan of not requiring more data as a prop than needed for a component to work, so since we can already get a module's name from inside the component when the `slug` is known and we are already selecting other parts of the module's info from state ([like its `icon`](https://github.com/google/site-kit-wp/pull/2808/files#diff-7214c466b8f8c31951bc809134e9dcaad0ee7a61a96c860a838721f8ca891141R51)), I think it'd be better to get the `name` from a `useSelect` instead of as a prop.

Reply:
But the selector will return `undefined` at first, no?

<img width=400 src=https://user-images.githubusercontent.com/2470911/108343795-459f3800-71d4-11eb-9fae-4f26345de547.png />